app/src/ui/discard-changes/discard-selection-dialog.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react'
+
+import { Repository } from '../../models/repository'
+import { Dispatcher } from '../dispatcher'
+import { WorkingDirectoryFileChange } from '../../models/status'
+import { Dialog, DialogContent, DialogFooter } from '../dialog'
+import { PathText } from '../lib/path-text'
+import { Monospaced } from '../lib/monospaced'
+import { Checkbox, CheckboxValue } from '../lib/checkbox'
+import { OkCancelButtonGroup } from '../dialog/ok-cancel-button-group'
+import { ITextDiff, DiffSelection } from '../../models/diff'
+
+interface IDiscardSelectionProps {
+  readonly repository: Repository
+  readonly dispatcher: Dispatcher
+  /**
+   * The file where the selection of changes to discard should be applied.
+   */
+  readonly file: WorkingDirectoryFileChange
+  /**
+   * The current diff with the local changes for that file.
+   */
+  readonly diff: ITextDiff
+  /**
+   * The selection (based on the passed diff) of changes to discard.
+   */
+  readonly selection: DiffSelection
+  /**
+   * Function called when the user either dismisses the dialog or
+   * the discard operation finishes.
+   */
+  readonly onDismissed: () => void
+}
+
+interface IDiscardSelectionState {
+  /**
+   * Whether or not we're currently in the process of discarding
+   * changes. This is used to display a loading state
+   */
+  readonly isDiscardingSelection: boolean
+  /**
+   * Whether or not the "do not show this message again" checkbox
+   * is checked.
+   */
+  readonly confirmDiscardSelection: boolean
+}
+
+/** A component to confirm and then discard changes from a selection. */
+export class DiscardSelection extends React.Component<
+  IDiscardSelectionProps,
+  IDiscardSelectionState
+> {
+  public constructor(props: IDiscardSelectionProps) {
+    super(props)
+
+    this.state = {
+      isDiscardingSelection: false,
+      confirmDiscardSelection: false,
+    }
+  }
+
+  private getOkButtonLabel() {
+    return __DARWIN__ ? 'Discard changes' : 'Discard changes'
+  }
+
+  public render() {
+    const isDiscardingChanges = this.state.isDiscardingSelection
+
+    return (
+      <Dialog
+        id="discard-changes"
+        title={
+          __DARWIN__ ? 'Confirm Discard changes' : 'Confirm Discard changes'
```

Comment:
```suggestion
          __DARWIN__ ? 'Confirm Discard changes' : 'Confirm discard changes'
```

Reply:
Ups thanks for noticing this!

I copy-pasted both strings from the `DiscardChanges` dialog: https://github.com/desktop/desktop/blob/development/app/src/ui/discard-changes/discard-changes-dialog.tsx#L63

I'm going to fix these as well!